app/src/main/java/org/traccar/client/DatabaseHelper.java

```diff
@@ -91,6 +91,11 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
         onCreate(db);
     }
 
+    public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        db.execSQL("DROP TABLE IF EXISTS position;");
```

Comment:
Pretty nasty to just delete data... :/

Reply:
How can you downgrade an app?

Reply:
From e.g. https://apkpure.com/traccar-client/org.traccar.client ?

Reply:
I guess you meant to say that you have to manually download APK. I don't think very many people do that.

As for the original comment, ideally database should be empty anyway if you have internet connection. Traccar Client uploads data automatically and removes it from local database.

Reply:
Right, regular Google Play Store users can't downgrad. But as can be seen @luke-jr did, so everything can happen.

Reply:
As for the original comment, I'm also fine with the way upgrading was done.

Reply:
I did it by copying the apk off another phone that hadn't upgraded yet.